packages/sns/candid/sns_index.did

```diff
@@ -0,0 +1,76 @@
+// Generated from IC repo commit f818cf1eb89bcbc9aecb1fb73de8c8de6c258c42 'rs/rosetta-api/icrc1/index/index.did' by import-candid
+type TxId = nat;
+
+type Account = record { owner : principal; subaccount : opt blob };
+
+type SubAccount = blob;
+
+type Transaction = record {
+     kind : text;
+     mint : opt record {
+         amount : nat;
+         to : Account;
+         memo : opt blob;
+         created_at_time : opt nat64;
+     };
+     burn : opt record {
+         amount : nat;
+         from : Account;
+         memo : opt blob;
+         created_at_time : opt nat64;
+     };
+     transfer : opt record {
+         amount : nat;
+         from : Account;
+         to : Account;
+         memo : opt blob;
+         created_at_time : opt nat64;
+     };
+     timestamp : nat64;
+};
+
+type GetAccountTransactionsArgs = record {
+    account : Account;
+    // The txid of the last transaction seen by the client.
+    // If None then the results will start from the most recent
+    // txid.
+    start : opt TxId;
+    // Maximum number of transactions to fetch.
+    max_results : nat;
+};
+
+type TransactionWithId = record {
+  id : TxId;
+  transaction : Transaction;
+};
+
+type GetTransactions = record {
+  transactions : vec TransactionWithId;
+  // The txid of the oldest transaction the account has
+  oldest_tx_id : opt TxId;
+};
+
+type GetTransactionsErr = record {
+  message : text;
+};
+
+type GetTransactionsResult = variant {
+  Ok : GetTransactions;
+  Err : GetTransactionsErr;
+};
+
+type ListSubaccountsArgs = record {
+    owner: principal;
+    start: opt SubAccount;
+};
+
+// The initialization parameters of the Index canister.
+type InitArgs = record {
+    ledger_id : principal;
+};
+
+service : (InitArgs) -> {
+  get_account_transactions : (GetAccountTransactionsArgs) -> (GetTransactionsResult);
```

Comment:
This is a good question. I'm surprise it's not a query. Let me ask about it in the SNS channel.

Reply:
But I believe it's because the endpoint makes calls to the ledger canister to get the data from there. That might be the reason why we can't do a query. But it's anyway worth asking.